src/linker/Linker.Steps/CodeRewriterStep.cs

```diff
@@ -109,27 +110,90 @@ MethodBody CreateStubBody (MethodDefinition method)
 			switch (method.ReturnType.MetadataType) {
 			case MetadataType.Void:
 				break;
-			case MetadataType.Boolean:
-				il.Emit (OpCodes.Ldc_I4_0);
-				break;
 			default:
-				throw new NotImplementedException (method.ReturnType.FullName);
+				var instruction = StubMethodWithConstant (Context, method);
+				if (instruction != null) {
+					il.Append (instruction);
+				} else {
+					StubComplexBody (method, body);
+				}
+				break;
 			}
 
 			il.Emit (OpCodes.Ret);
 			return body;
 		}
 
-		MethodBody CreateReturnFalseBody (MethodDefinition method)
+		static void StubComplexBody (MethodDefinition method, MethodBody body)
 		{
-			if (method.ReturnType.MetadataType != MetadataType.Boolean)
-				throw new NotImplementedException ();
+			switch (method.ReturnType.MetadataType) {
+			case MetadataType.MVar:
+			case MetadataType.ValueType:
+				var vd = new VariableDefinition (method.ReturnType);
+				body.Variables.Add (vd);
+				body.InitLocals = true;
+
+				var il = body.GetILProcessor ();
+				il.Emit (OpCodes.Ldloca_S, vd);
+				il.Emit (OpCodes.Initobj, method.ReturnType);
+				il.Emit (OpCodes.Ldloc_0);
+				return;
+			}
 
-			var body = new MethodBody (method);
-			var il = body.GetILProcessor ();
-			il.Emit (OpCodes.Ldc_I4_0);
-			il.Emit (OpCodes.Ret);
-			return body;
+			throw new NotImplementedException (method.ReturnType.MetadataType.ToString ());
```

Comment:
I think it would be nice to display the name of the method that was beings stubbed in the exception message.

Reply:
Changed